File: Resueltos_por_competidor/Juancho_Garcia/GoodContest.java
/*
 * Autor: Juan Andres Garcia
 * Problema: Good Contest
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/contest/2266/problem/A
 *  **/


import java.util.Scanner;

public class GoodContest {
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		StringBuilder sb = new StringBuilder();
		
		
		int T = sc.nextInt(); 
		
		for(int i = 1; i <=T; i++) {
			
			int participantes = sc.nextInt();
			
			
			int p1 = sc.nextInt();
			int p2 = sc.nextInt();
			int p3 = sc.nextInt();
			
			int coso = 0;
			int result = 0;
			
			
			coso = Math.min(p1, Math.min(p2, p3));
			
			
			result = participantes - coso;
			
			
			sb.append(result+"\n");
			
				
			
			
			
		}
		
		
		System.out.println(sb.toString());
		
		
	}

}

File: Resueltos_por_competidor/Juan_Martinez/GoodContest.cpp
/*
 * Autor: Juan Martinez
 * Problema: Good Contest (2266A)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/contest/2266/problem/A
 */
#include <bits/stdc++.h>
using namespace std;

#define ln "\n"
#define vvi vector<vector<int>>
#define vvl vector<vector<ll>>
#define vi vector<int>
#define vl vector<ll>
#define vb vector<bool>
#define pii pair<int,int>
#define pl pair<ll,ll>
#define vpii vector<pii>
#define vpl vector<pl>
#define add(x) push_back(x)
const int INF = 1e9;

int main() {
	int t, n; 
    cin>>t;
    while(t--) {
        int arr[3];
        cin>>n;
        int mini = INF, maxi = 0;
        for(int i = 0; i < 3; i++) {
            cin>>arr[i];
            mini = min(mini, arr[i]);
            maxi = max(maxi, arr[i]);
        }
        cout<<(n-maxi) + (maxi-mini)<<ln;
    }
}
